### ISSUE TITLE
BoxyChild receives null parentData on first layout

### ISSUE THREAD

**Issue by MikeFP** [PixelToast/flutter-boxy#9]
The first time `BoxyDelegate.layout()` is processed, trying to access a child's `parentData` evaluates to `null`.

After a hot reload or changing the screen state to rebuild it, layout is called again but parentData is now accessible.

Code example:

```
import 'package:boxy/boxy.dart';
import 'package:flutter/widgets.dart';

class TestBug extends StatelessWidget {
  const TestBug({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      child: CustomBoxy(
        delegate: MyBoxy(),
        children: [
          BoxyId<String>(
            id: 'child-0',
            data: "I'm a child",
            child: SizedBox(width: 50, height: 50),
          ),
        ],
      ),
    );
  }
}

class MyBoxy extends BoxyDelegate {
  @override
  Size layout() {
    final child = getChild('child-0');
    print(child.parentData);
    print(child.parentData as String?);
    child.layout(constraints);
    child.position(Offset.zero);
    return constraints.biggest;
  }
}
```

Putting this widget in a Scaffold's body prints null twice, which is unexpected. Hitting Ctrl+S to save the file triggers a hot reload and prints "I'm a child" twice, as expected.

**Comment by PixelToast** [comment]
I reproduced the issue and published a fix in version 2.0.4, let me know if that works!

**Comment by MikeFP** [comment]
@PixelToast It's working fine. Thanks!
